File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head==None or head.next==None:
            return head
        prev=head
        crr=head.next
        while crr!=None:
            if prev.val==crr.val:
                crr=crr.next
            else:
                prev.next=crr
                prev=crr
                crr=crr.next
        prev.next=None
        return head

File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.java

class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        
        if(head==null || head.next==null){
            return head;
        }

        ListNode prev=head;
        ListNode crr=head.next;
        while(crr!=null){
            if(prev.val==crr.val){
                crr=crr.next;
            }
            else{
                prev.next=crr;
                prev=crr;
                crr=crr.next;
            }
        }

        prev.next=null;

        return head;
    }
}